### PR TITLE
react1-week2 homework

### PR DESCRIPTION
**Pull request by shravanipagilla**

### CODE REVIEW

File: react/week2/my-react-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from "react";
+import Todos from './Todos/Todos';
+import AddTodo from './Todos/AddTodo';
+import './App.css';
+
+const arrayOfTodos = [
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/App.js]: you can store this data in another file

File: react/week2/my-react-app/src/App.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+function App() {
+  const [todos, setTodos] = useState(arrayOfTodos);
+ 
+  //adding todo
+  const addTodo = (description) => {
+    if (description) {
+      const id = Math.floor(Math.random() * (100 - 4 + 1)) + 4;
+      console.log(id)
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/App.js]: clean you console.logs

File: react/week2/my-react-app/src/App.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  )
+}
+
+function App() {
+  const [todos, setTodos] = useState(arrayOfTodos);
+ 
+  //adding todo
+  const addTodo = (description) => {
+    if (description) {
+      const id = Math.floor(Math.random() * (100 - 4 + 1)) + 4;
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/App.js:13]: smart way of making a unique id! 😉

File: react/week2/my-react-app/src/App.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      setTodos(todos.concat({
+        id: id,
+        description: description
+      }
+      ))
+    }
+  }
+  //deleting todo
+  const deleteTodo = (id) => {
+    console.log("delete", id)
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/App.js]: clean up console.logs

File: react/week2/my-react-app/src/Todos/AddTodo.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            setVisible(false);
+            onAdd(value);
+        }
+
+        setValue("")
+    }
+    return (
+        <div>
+            {visible && <input type="text" className="input-text" placeholder="Enter description" onChange={(e => setValue(e.target.value))} ></input>}
+            <button onClick={onClick}>{visible ? "Submit" : "Add new Todo"}</button>
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/Todos/AddTodo.jsx]: submit has to work with Enter key

File: react/week2/my-react-app/src/App.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  return (
+      <div>You are using this webpage from {counter} seconds</div>
+  )
+}
+
+function App() {
+  const [todos, setTodos] = useState(arrayOfTodos);
+ 
+  //adding todo
+  const addTodo = (description) => {
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/App.js]: you have a function and a component called addTodo, it's confusing

File: react/week2/my-react-app/src/App.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from "react";
+import Todos from './Todos/Todos';
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/App.js]: the components folder is usually called components

File: react/week2/my-react-app/src/Todos/Todos.jsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import React, { useState } from "react";
+
+const TodoItem = ({ id, description, onDelete }) => {
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/Todos/Todos.jsx]: this should be another component

File: react/week2/my-react-app/src/Todos/AddTodo.jsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import React, { useState } from "react";
+// @ts-ignore
+
+const AddTodo = ({ onAdd }) => {
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/Todos/AddTodo.jsx]: this whole code should be in TodoItem component

File: react/week2/my-react-app/src/Todos/Todos.jsx [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+        </div>
+    )
+}
+
+const Todos = ({ todoArray, onDelete }) => {
+    return (
+        <ul>
+            {todoArray.map((todo) => {
+                return <li key={todo.id}>
+                    <TodoItem id={todo.id} description={todo.description} onDelete={onDelete} />
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/Todos/Todos.jsx]: id and description should be accessed only on TodoItem

File: react/week2/my-react-app/src/App.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+function App() {
+  const [todos, setTodos] = useState(arrayOfTodos);
+ 
+  //adding todo
+  const addTodo = (description) => {
+    if (description) {
+      const id = Math.floor(Math.random() * (100 - 4 + 1)) + 4;
+      console.log(id)
+      setTodos(todos.concat({
```
Comment (annamourasobral) [inline react/week2/my-react-app/src/App.js:14]: if you are using setTodos to add a new item you can simply destructure the array and add the new todo 
```suggestion
      setTodos(...todos, {
```

**Review by annamourasobral — CHANGES_REQUESTED** [review]
Hey, @shravanipagilla 
Really nice code, with a lot of high-order functions, props, and ternary operators, you are working with real-life developers' tools! 
I suggested some adjustments to get it even better, I hope you find it useful. 
Happy coding! 👩🏻‍💻

File: react/week2/my-react-app/src/App.js (hunk 3 above)
Comment (shravanipagilla) [reply react/week2/my-react-app/src/App.js:13]: thank you :)

**Comment by annamourasobral** [comment]
Hey, @shravanipagilla 
I saw that you delivered PR #31 with the same suggestions that I commented on here. I greatly recommend you address them before going forward next time.
It spares you from doing the same adjustments in two different PR and the Mentors from reviewing the same code.
If you have any questions about the review don't hesitate to contact me! ☺️

**Comment by shravanipagilla** [comment]
> Hey, @shravanipagilla I saw that you delivered PR #31 with the same suggestions that I commented on here. I greatly recommend you address them before going forward next time. It spares you from doing the same adjustments in two different PR and the Mentors from reviewing the same code. If you have any questions about the review don't hesitate to contact me! ☺️

Hej @annamourasobral thank you and sorry for that. with react my mind is not working properly... I made changes as per the comments some of not understand I will do it later ... thank you for your time:)

**Review by annamourasobral — APPROVED** [review]
Thank you for the updates!

**Pull request merged**